tests/src/testfebases.cpp

```diff
@@ -0,0 +1,164 @@
+
+#include <dune/common/test/testsuite.hh>
+#include <dune/functions/functionspacebases/basistags.hh>
+#include <dune/functions/functionspacebases/compositebasis.hh>
+#include <dune/functions/functionspacebases/lagrangebasis.hh>
+#include <dune/functions/functionspacebases/lagrangedgbasis.hh>
+#include <dune/functions/functionspacebases/nedelecbasis.hh>
+#include <dune/functions/functionspacebases/powerbasis.hh>
+#include <dune/functions/functionspacebases/raviartthomasbasis.hh>
+#include <dune/grid/yaspgrid.hh>
+
+using Dune::TestSuite;
+
+#include <ikarus/finiteelements/febases.hh>
+#include <ikarus/finiteelements/fehelper.hh>
+#include <ikarus/utils/basis.hh>
+#include <ikarus/utils/init.hh>
+
+/**
+ * \brief A helper function to get a FEBase.
+ *
+ * \tparam GridView Type of the grid view
+ * \tparam PreBasis Type of the pre basis
+ * *
+ * \param gridView The DUNE grid view.
+ * \param preBasis The pre basis to make a basis.
+ *
+ * \return FEBase corresponding to the gridView and preBasis.
+ */
+template <typename GridView, typename PreBasis>
+auto getBasis(const GridView& gridView, const PreBasis& preBasis) {
+  const auto basis      = Ikarus::makeBasis(gridView, preBasis);
+  const auto& localView = basis.flat().localView();
+  auto element          = elements(gridView).begin();
+  return Ikarus::FEBase<decltype(basis)>(basis, *element);
+}
+
+/**
+ * \brief A test suite to test a set of bases.
+ * \tparam numberOfBases Number of bases to be tested.
+ * \tparam Bases Type of different bases.
+ * \param expectedNumberOfChildren Expected number of children for each basis.
+ * \param expectedSize Expected size for each basis.
+ * \param bases A tuple of bases.
+ * \param basisName Name of the types of bases.
+ * \return Test suite object.
+ */
+template <size_t numberOfBases, typename... Bases>
+auto BasisTest(const std::array<int, numberOfBases>& expectedNumberOfChildren,
+               const std::array<int, numberOfBases>& expectedSize, const std::tuple<Bases...>& bases,
+               const std::string& basisName) {
+  static_assert(std::tuple_size_v<std::tuple<Bases...>> == numberOfBases, "Input size mismatch in BasisTest.");
+  TestSuite t("BasisTest");
+  Dune::Hybrid::forEach(Dune::Hybrid::integralRange(Dune::index_constant<numberOfBases>()), [&](const auto i) {
+    const auto basis          = std::get<i>(bases);
+    const std::string message = " for i = " + std::to_string(i) + " -> " + basisName;
+    using MultiIndex          = typename std::remove_cvref_t<decltype(basis.localView())>::MultiIndex;
+    static_assert(Dune::IsIndexable<MultiIndex>(), "MultiIndex must support operator[]");
+    checkScalars(t, basis.numberOfChildren(), expectedNumberOfChildren[i],
+                 " Number of children is incorrect" + message);
+    checkScalars(t, static_cast<int>(basis.size()), expectedSize[i], " Size is incorrect" + message);
+    std::vector<MultiIndex> dofs;
+    Ikarus::FEHelper::globalFlatIndices(basis.localView(), dofs);
```

Comment:
Actually `globalFlatIndices` would exactly look the same for blocked basis. We should rename the function to globalIndices. Can you also provide examples fro the blocked case? Yo simply have to make the getBasis function above with untouched instead of flat.